### PR TITLE
Migrate tab.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/ui/tab.dart [hunk 1]
```diff
@@ -19,13 +19,13 @@ class DevToolsTab extends Tab {
   /// The only difference is this tab makes more of an effort to reflect
   /// changes in font and icon sizes.
   DevToolsTab._({
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/tab.dart:21]: make key and gaid required

File: packages/devtools_app/lib/src/ui/tab.dart [hunk 2]
```diff
@@ -63,9 +63,9 @@ class DevToolsTab extends Tab {
   }
 
   /// Tab id for google analytics.
-  final String gaId;
+  final String? gaId;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/tab.dart:66]: non-nullable

File: packages/devtools_app/lib/src/ui/tab.dart [hunk 3]
```diff
@@ -165,8 +165,8 @@ class _AnalyticsTabbedViewState extends State<AnalyticsTabbedView>
 
   @override
   void dispose() {
-    _tabController?.removeListener(_onTabChanged);
-    _tabController?.dispose();
+    _tabController!.removeListener(_onTabChanged);
+    _tabController!.dispose();
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/tab.dart:169]: leave the ? null checks here instead of force unwrapping

Comment (polina-c) [reply packages/devtools_app/lib/src/ui/tab.dart:169]: done

File: packages/devtools_app/lib/src/ui/tab.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/ui/tab.dart:21]: done

File: packages/devtools_app/lib/src/ui/tab.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/ui/tab.dart]: done

File: packages/devtools_app/lib/src/ui/tab.dart [hunk 4]
```diff
@@ -19,13 +19,13 @@ class DevToolsTab extends Tab {
   /// The only difference is this tab makes more of an effort to reflect
   /// changes in font and icon sizes.
   DevToolsTab._({
-    Key key,
-    String text,
-    Icon icon,
+    required Key? key,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/tab.dart:22]: required and non-nullable. The only place this should be constructed is from the factory constructor, and we know it is non-null.

**Review by kenzieschmoll — APPROVED** [review]
one comment then lgtm

**Comment by polina-c** [comment]
tested

File: packages/devtools_app/lib/src/ui/tab.dart (hunk 4 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/ui/tab.dart]: done

**Comment by polina-c** [comment]
tested

**Pull request merged**